nebula/ui/components/inAppAuth/VPNInAppAuthenticationInputs.qml

```diff
@@ -46,24 +46,61 @@ ColumnLayout {
 
         spacing: VPNTheme.theme.listSpacing
 
-        VPNTextField {
-            id: textInput
-            objectName: base.objectName + "-textInput"
+        RowLayout {
+            id: textInputSection
             Layout.fillWidth: true
-            _placeholderText: _inputPlaceholderText
-            Keys.onReturnPressed: col.submitInfo(textInput)
-            onDisplayTextChanged: if (hasError) hasError = false
+            spacing: VPNTheme.theme.windowMargin / 2
+
+            VPNTextField {
+                id: textInput
+                objectName: base.objectName + "-textInput"
+                Layout.fillWidth: true
+                _placeholderText: _inputPlaceholderText
+                Keys.onReturnPressed: col.submitInfo(textInput)
+                onDisplayTextChanged: if (hasError) hasError = false
+            }
+
+            VPNPasteButton {
+                id: textInputPasteButton
+                objectName: base.objectName + "-textInputPasteButton"
+                Layout.preferredWidth: VPNTheme.theme.rowHeight
+                Layout.preferredHeight: VPNTheme.theme.rowHeight
+                height: undefined
+                width: undefined
+                onClicked: {
+                    textInput.paste();
```

Comment:
Do you want to make the button disabled if there is nothing to paste?

Reply:
**Yes.** I talked with Santiago and Peter about hiding it or disabling and the feedback was to leave it visible and enabled out of concern that we may incorrectly hide/show or disable/enable it based on faulty clipboard access on Android. Now that this seems to be largely resolved by the changes over in `VPNActivity.java` I think we should reconsider this decision. I'll take it up with them again next week.

Reply:
.